pygfx/renderers/wgpu/meshshader.py

```diff
@@ -68,9 +68,18 @@ def get_bindings(self, wobject, shared):
         if getattr(geometry, "normals", None) is not None:
             normal_buffer = geometry.normals
         else:
-            normal_data = normals_from_vertices(
-                geometry.positions.data, geometry.indices.data
-            )
+            if geometry.indices.data.shape[1] == 4:
+                n1 = normals_from_vertices(
+                    geometry.positions.data, geometry.indices.data[:,:3]
+                )
+                n2 = normals_from_vertices(
+                    geometry.positions.data, geometry.indices.data[:,[0,2,3]]
+                )
+                normal_data = (n1+n2)/2
```

Comment:
do you really need to compute the normals for both triangles of every quad though? aren't they always going to be equal for each triangle that makes up the quad?

I'd say you can just compute the normals for one triangle and work with that, right?

Reply:
I am averaging them because of this post:
https://stackoverflow.com/questions/5410143/calculating-face-normals-with-4-vertices

The quads in my mesh are not planar so I average them.

Reply:
I never realized such quad geometry existed, I always assumed they would be planar! In that case, it's possible the two triangles have different surface area, correct? Should it be a weighted average to correct for that, or is the error assumed to be very small and thus can it be safely ignored?

Reply:
After looking at this again, it's already correct.

Reply:
Yes the area of both the triangles are different. These are irregular convex quads so the normals can be a bit different in both the triangles. I read that cross product of the diagonals would be the most apt but I haven't tested that.

Reply:
I played around with this, there is an improvement to be made, I added a separate suggestion comment below.